quinn-proto/src/config.rs

```diff
@@ -575,6 +581,12 @@ impl ClientConfig {
         }
     }
 
+    /// Set a custom [`TransportConfig`]
+    pub fn with_transport_config(mut self, transport: TransportConfig) -> Self {
+        self.transport = Arc::new(transport);
+        self
+    }
```

Comment:
The `with_` prefix is used below for constructors, rather than setters, which seems a bit confusing. We're also using by-`&mut` builders everywhere else.